src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.goide.psi.impl.GoPsiImplUtil;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Implements <a href="https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1983"/>, an
+ * inspector that warns if a go StructTag is not well-formed according to Go language conventions.
+ */
+public class GoStructInitializationInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitCompositeLit(@NotNull GoCompositeLit o) {
+        super.visitCompositeLit(o);
+
+        if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) return;
+        GoType type = o.getType();
+        if (!(type instanceof GoStructType) &&
+            !(GoPsiImplUtil.findBaseTypeFromRef(o.getTypeReferenceExpression()) instanceof GoSpecType)) {
+          return;
+        }
+
+        for (GoElement element : o.getLiteralValue().getElementList()) {
+          if (element.getKey() == null) {
+            holder.registerProblem(o, "Structure has unnamed field initialization", ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
```

Comment:
In that case, the compiler will throw an error.
In this case things will work, but if there's a refactoring done, they will break. That's why I've made this as a weak warning and not an warning / error. 
I've also mixed-up the PRs. for this a quickfix would be to add the field names, however since one can only partially initialize the struct, we don't know which field to use (maybe display a list of fields is a good idea then).